model/src/w3srcemd.F90

```diff
@@ -1378,9 +1395,123 @@ SUBROUTINE W3SRCE ( srce_call, IT, JSEA, IX, IY, IMOD,          &
                WRITE(740+IAPROC,*) 'min/max/sum(VDBT)=', minval(VDBT), maxval(VDBT), sum(VDBT)
         END IF
 #endif
+         IF (srce_call .eq. srce_imp_pre) THEN
+#ifdef W3_PDLIB
+      IF (LSLOC) THEN
+        IF (IMEM == 1) THEN
+          SIDT  = PDLIB_SI(JSEA) * DTG 
+          DO IK = 1, NK
+            JAC = CLATSL/CG1(IK) 
+            DO ITH = 1, NTH
+              ISP = ITH + (IK-1)*NTH
+#endif
```

Comment:
This #endif and the following #ifdef W3_PDLIB is not necessary

Reply:
done